src/js/utils/linked-list.js

```diff
@@ -11,9 +15,15 @@ export default class LinkedList {
     }
   }
   adoptItem(item) {
+    item[PARENT_PROP]= this;
+    this.length++;
     if (this._adoptItem) { this._adoptItem(item); }
   }
   freeItem(item) {
+    item[PARENT_PROP] = null;
+    item.prev   = null;
+    item.next   = null;
```

Comment:
Ah you've moved it. As `prev` and `next` are related to position and not ownerships (`adopt`/`free`) IMO the changes should be made in remove.